### PR TITLE
Implement run-time backtraces and other improvements

### PR DESCRIPTION
**Pull request by housel**
These changes implement printing a backtrace when a Dylan program is terminated with an error condition, and make other improvements that improve platform support.

### CODE REVIEW

File: sources/lib/run-time/demangle.c [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#define MODULE_SEP              'Y'
+#define LIBRARY_SEP             'V'
+#define ESCAPE_SEP              'Z'
+#define METHOD_MARK             'M'
+#define IEP_MARK                'I'
+
+#define ARRAY_LEN(x)            (sizeof(x)/sizeof((x)[0]))
+
+// This is adequte for IEPs, but could be improved for other things
+int dylan_demangle(char *dest, size_t destsize, char *src)
```
Comment (cgay) [inline sources/lib/run-time/demangle.c]: How about linking to https://opendylan.org/documentation/hacker-guide/runtime/mangling.html here?

File: sources/lib/run-time/demangle.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            *dest++ = *library++;
+          }
+          module_seen = true;
+        }
+        if (c == METHOD_MARK) {
+          *dest++ = '#';
+        }
+        break;
+
+        break;
```
Comment (cgay) [inline sources/lib/run-time/demangle.c]: Extraneous break statement?

File: sources/lib/run-time/stack-walker.c [hunk 3]
```diff
@@ -1,71 +1,299 @@
...
+
+      fprintf(stderr, "  %#jx\n", (uintmax_t) ip);
+    }
+
+    // On to the next enclosing frame
+    rc = unw_step(&cursor);
+  } while (rc > 0);
 }
 
 #else
```
Comment (cgay) [inline sources/lib/run-time/stack-walker.c]: Add a comment since so far away from the #ifdef?

**Comment by pedro-w** [comment]
I had a look at this by starting from the master branch of dylan-lang/opendylan and pulling from your branch run-time-backtrace-1. I was using debian 9 64-bit with clang-7 installed.
`make check` was OK (no fails that don't happen for master as far as I can see)
When I installed it and built my own app, created with `make-dylan-app cr`, built with `dylan-compiler -back-end llvm -build cr` linking failed at the last step, citing missing functions like `sin`, `cos` from `libcommon-dylan.so`.
I added `-lm` to x86_64-linux-build.jam and it was OK.
I tried to trigger a back-trace but never saw anything, is this right?
``` dylan
Module: cr
Synopsis: 
Author: 
Copyright: 

define function it()
  error ("here is an error")
end function;
  
define function main
  (name :: <string>, arguments :: <vector>)
  format-out("We're OK,");
  it();
  format-out("Hello, world!\n");
  exit-application(0);
end function main;

main(application-name(), application-arguments());
```
output

    $ ./cr
    here is an error

**Comment by pedro-w** [comment]
OK, as I mentioned on slack - my fault. Need to re-run autogen.sh before configure and make.

**Pull request merged**